### ISSUE TITLE
Certain queries involving STRUCT cause BE crash

### ISSUE THREAD

**Issue by rcauble** [StarRocks/starrocks#32061]
### Steps to reproduce the behavior (Required)
unzip [bug.zip](https://github.com/StarRocks/starrocks/files/12822165/bug.zip)
cd bug-repro
./bug.sh

### Expected behavior (Required)
Should not crash

### Real behavior (Required)
ERROR 1064 (HY000) at line 2: Backend node not found. Check if any backend node is down.backend: [d0a083668256 alive: true inBlacklist: true] 

### StarRocks version (Required)
starrocks/allin1-ubuntu:3.1.0-rc01

**Comment by fzhedu** [comment]
set enable_prune_complex_types=false
can disable it, and we will fixed the bug ASAP.

**Comment by rcauble** [comment]
The suggested workaround reduces the frequency of the problem, but I still see the crash and/or evidence of memory corruption sometimes in other queries. Here's a a new bug.zip with a different query and bug.sh script:

[bug.zip](https://github.com/StarRocks/starrocks/files/12849490/bug.zip)

In this case it runs the same query 11 times. All runs should produce 0 diffs vs expected. But as you can see, some of them produce a diff and others do not. I think this is likely due to some memory corruption. Here's an example output that I see: 

```
./bug.sh 
24604e7c6c5c8aa699e7707302d9cc0492ee843e0cf07ea3c26d5c77f215f2bd
Sleeping for 1 minute
Creating database
{
    "TxnId": 2,
    "Label": "46b49bd6-f6a0-4661-bf82-d0fcc7789757",
    "Status": "Success",
    "Message": "OK",
    "NumberTotalRows": 11,
    "NumberLoadedRows": 11,
    "NumberFilteredRows": 0,
    "NumberUnselectedRows": 0,
    "LoadBytes": 1418,
    "LoadTimeMs": 159,
    "BeginTxnTimeMs": 21,
    "StreamLoadPlanTimeMs": 55,
    "ReadDataTimeMs": 0,
    "WriteDataTimeMs": 42,
    "CommitAndPublishTimeMs": 39
}{
    "TxnId": 3,
    "Label": "a228817a-6d27-48d2-83ce-8d12b779fc6b",
    "Status": "Success",
    "Message": "OK",
    "NumberTotalRows": 4,
    "NumberLoadedRows": 4,
    "NumberFilteredRows": 0,
    "NumberUnselectedRows": 0,
    "LoadBytes": 1724,
    "LoadTimeMs": 63,
    "BeginTxnTimeMs": 0,
    "StreamLoadPlanTimeMs": 5,
    "ReadDataTimeMs": 0,
    "WriteDataTimeMs": 34,
    "CommitAndPublishTimeMs": 22
}Running Bug.sql 1
1,2c1,2
< 10	2	Ctest2	Ctest1	1
< 11	2	Ctest2	Ctest1	1
---
> 10	2	Btest2	Btest1	1
> 11	2	Btest2	Btest1	1
13,14d12
< NULL	2	Btest2	Btest1	1
< NULL	2	Btest2	Btest1	1
Running Bug.sql 2
1,2c1,2
< 10	2	Ctest2	Ctest1	1
< 11	2	Ctest2	Ctest1	1
---
> 10	2	Btest2	Btest1	1
> 11	2	Btest2	Btest1	1
13,14d12
< NULL	2	Btest2	Btest1	1
< NULL	2	Btest2	Btest1	1
Running Bug.sql 3
Running Bug.sql 4
1,2c1,2
< 10	2	Ctest2	Ctest1	1
< 11	2	Ctest2	Ctest1	1
---
> 10	2	Btest2	Btest1	1
> 11	2	Btest2	Btest1	1
13,14d12
< NULL	2	Btest2	Btest1	1
< NULL	2	Btest2	Btest1	1
Running Bug.sql 5
Running Bug.sql 6
1,2c1,2
< 10	2	Ctest2	Ctest1	1
< 11	2	Ctest2	Ctest1	1
---
> 10	2	Btest2	Btest1	1
> 11	2	Btest2	Btest1	1
13,14d12
< NULL	2	Btest2	Btest1	1
< NULL	2	Btest2	Btest1	1
Running Bug.sql 7
Running Bug.sql 8
Running Bug.sql 9
1,2c1,2
< 10	2	Ctest2	Ctest1	1
< 11	2	Ctest2	Ctest1	1
---
> 10	2	Btest2	Btest1	1
> 11	2	Btest2	Btest1	1
13,14d12
< NULL	2	Btest2	Btest1	1
< NULL	2	Btest2	Btest1	1
Running Bug.sql 10
Running Bug.sql 11
1,2c1,2
< 10	2	Ctest2	Ctest1	1
< 11	2	Ctest2	Ctest1	1
---
> 10	2	Btest2	Btest1	1
> 11	2	Btest2	Btest1	1
13,14d12
< NULL	2	Btest2	Btest1	1
< NULL	2	Btest2	Btest1	1
```

In this case, runs 1, 2, 4, 6, 9, and 11 produced a diff. Either all of them or none of them should produce a diff since it's the same query in all cases.

**Comment by fzhedu** [comment]
after https://github.com/StarRocks/starrocks/pull/32393, sr produces stable results and diffs from the expected results.

I wonder whether the expected results in  `expected.txt` is right? from the first with clause, we see the the last line ` ["C","B","B"]                      | [9,10,11]`, so if unnesting, 10 -> B and 11 -> B, they cann't result 10 -> Ctest2 and 11-> Ctest2.

```
mysql>   SELECT
    ->     `TableScan_Alias_CASE_000`.`ID_CASE` AS `ID_CASE`,
    ->     `TableScan_Alias_CASE_000`.`PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITYNAME` AS `TableGroupNodeHandler_argument_000`,
    ->     `TableScan_Alias_CASE_000`.`PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITY` AS `PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITY`,
    ->     ARRAY_MAP(
    ->       (indexArg_1) -> indexArg_1 -1,
    ->       ARRAY_CUM_SUM(
    ->         ARRAY_MAP(
    ->           (indexArg_1) -> CAST(1 AS BIGINT),
    ->           `TableScan_Alias_CASE_000`.`PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITYNAME`
    ->         )
    ->       )
    ->     ) AS `PU_9_ARRAY_AGG_8_ACTIVITY_INDEX_ACTIVITY`
    ->   FROM
    ->     `PREAGG_ARRAY_NEW_ALIGNMODEL_TRUE`.`CASE` AS `TableScan_Alias_CASE_000`;
+---------+------------------------------------+------------------------------------+------------------------------------------+
| ID_CASE | TableGroupNodeHandler_argument_000 | PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITY | PU_9_ARRAY_AGG_8_ACTIVITY_INDEX_ACTIVITY |
+---------+------------------------------------+------------------------------------+------------------------------------------+
|       4 | ["A","B","C"]                      | [3,4,5]                            | [0,1,2]                                  |
|       2 | ["A","C"]                          | [1,2]                              | [0,1]                                    |
|       6 | ["C","B","B"]                      | [6,7,8]                            | [0,1,2]                                  |
|       8 | ["C","B","B"]                      | [9,10,11]                          | [0,1,2]                                  |
+---------+------------------------------------+------------------------------------+------------------------------------------+
4 rows in set (0.09 sec)
```

please help me confirm the right results.

**Comment by rcauble** [comment]
It's entirely possible that the expected is wrong -- the point is that it should always produce the same results.

**Comment by fzhedu** [comment]
the origional issue is wrok aroud by disable enable_prune_complex_types, it should be fixed.

**Comment by Smith-Cruise** [comment]
fixed in 3.1.5
